pip_test.go

```diff
@@ -141,10 +141,12 @@ func testPipCmd(t *testing.T, projectPath, buildNumber, module string, expectedD
 func assertDependenciesRequestedByAndChecksums(t *testing.T, module buildinfo.Module, moduleName string) {
 	for _, dependency := range module.Dependencies {
 		assertDependencyChecksums(t, dependency.Checksum)
+		// Note that the versions of dependencies may vary,
+		// as they are requested by the main module, which could update their sub-dependencies.
```

Comment:
```suggestion
		// Note: Sub-dependency versions may vary because root dependencies can specify version ranges (e.g., >=1.0.0) for their sub-dependencies.
```